### PR TITLE
nf-test

### LINKED ISSUE #231: Setup nf-test for DSL2 Transition
Going to setup pytest to track tests and results to verify the DSL2 conversion, so we can have confidence in the reproduction and know of any changes.

Going to avoid setting up CI for them for now, and we can just run them manually locally to verify.

### PR DESCRIPTION
**Pull request by Emiller88**
closes #231

### CODE REVIEW

**Comment by Emiller88** [comment]
Need something like this that collects all the test results
https://github.com/pcolby/qtpokit/blob/main/.github/workflows/build.yaml

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ env:
 
 jobs:
   test:
-    name: Run pipeline with test data
+    name: Run ${{ matrix.aligner }}-${{ matrix.profile }} pipeline test
```
Comment (ewels) [inline .github/workflows/ci.yml:17]: Add NXF_VER here too?

Comment (ewels) [inline .github/workflows/ci.yml:17]:
```suggestion
    name: Run ${{ matrix.aligner }}-${{ matrix.profile }}-${{ matrix.NXF_VER }}
```

**Review by ewels — APPROVED** [review]
Looks amazing!

**Pull request merged**